### ISSUE TITLE
Simple program breaks compiler

### ISSUE THREAD

**Issue by lurk101** [HPCguy/Squint#47]
```
int main() {
    int a = 0;
    if (1) {
        int b;
        printf("%d\n", a);
        int c;
        b = 1;
        a += b;
        printf("%d\n", a);
    }
}
```
Results in
```
0
0
1
```
Clearly it should not print 3 numbers!

**Comment by HPCguy** [comment]
Technicall, this is a jserv/amacc bug, so plese submit the ticket there.  That said, they don't really fix anything over there anymore, so I will fix it here.  Nonetheless, when I see this issue show up there, I will fix it here.

**Comment by lurk101** [comment]
Thank you

**Comment by lurk101** [comment]
If it helps, the problem seems to be that gen is called twice for the 1st printf. I have not been able to track down the root cause. Moving all of the declaration to the top of their respective code blocks resolves the issue, but I thing the intent was to allow declarations to be positioned after executable statements.

**Comment by HPCguy** [comment]
Yes, that helps quite a lot, and would also help any developer immensely at jserv/amacc that may want to pick up this ticket.

**Comment by HPCguy** [comment]
Closed by #50
